sqlglot/dialects/fabric.py

```diff
@@ -65,6 +65,16 @@ class Generator(TSQL.Generator):
         }
 
         def datatype_sql(self, expression: exp.DataType) -> str:
+            # DATETIMEOFFSET is only supported when used with AT TIME ZONE:
+            # https://learn.microsoft.com/en-us/sql/t-sql/data-types/datetimeoffset-transact-sql#microsoft-fabric-support
+            if expression.is_type(exp.DataType.Type.TIMESTAMPTZ):
+                attimezone = expression.find_ancestor(exp.AtTimeZone, exp.Select)
+
+                if not attimezone:
```

Comment:
Minor oversight on my end here, fixed it really quick:
```suggestion
                if not isinstance(attimezone, exp.AtTimeZone):
```